src/lsp-server.ts

```diff
@@ -49,15 +50,27 @@ export class LspServer {
     }
 
     didChangeWorkspaceFolders(params: DidChangeWorkspaceFoldersParams) {
-        this.rootPath = fileURLToPath(params.event.added[0].uri);
-        this.runCtags(this.rootPath);
+        const added = params.event.added;
+        const removed = params.event.removed;
+        added.forEach(add => {
+            const rootPath = fileURLToPath(add.uri);
+            this.runCtags(rootPath);
+            this.rootPaths.push(rootPath);
+        });
+        removed.forEach(remove => {
+            const index = this.rootPaths.indexOf(fileURLToPath(remove.uri));
+            if (index !== -1) {
+                this.rootPaths.slice(index, 1);
```

Comment:
I'm not a ts expert. Do you mean `splice` here? `slice` method will return a new array, the original array will not be modified, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice.

Reply:
I should use `splice` instead of `slice` actually. And It's done now